backend/src/pong-game/entities/pong-game.entity.ts

```diff
@@ -0,0 +1,218 @@
+import { ObjectType, Field, Int, InputType } from '@nestjs/graphql'
+
+@ObjectType()
+class playfield {
+  @Field(() => Int)
+  width = 800
+
+  @Field(() => Int)
+  height = 600
+}
+
+@ObjectType()
+class racket {
+  constructor(hPos: number) {
+    this.hPos = hPos
+  }
+
+  @Field(() => Int)
+  width = 10
+
+  @Field(() => Int)
+  height = 60
+
+  @Field(() => Number)
+  hPos: number
+
+  @Field(() => Number)
+  vPos = 270
+
+  @Field(() => Int)
+  velocity = 6
+}
+
+@ObjectType()
+class ball {
+  @Field(() => Int)
+  radius = 16
+
+  @Field(() => Number)
+  hPos = 392
+
+  @Field(() => Number)
+  vPos = 292
+
+  dir: { x: number; y: number } = { x: 100, y: 100 }
+}
+
+@ObjectType()
+export class Controls {
+  @Field(() => Boolean)
+  Z_Key = false
+
+  @Field(() => Boolean)
+  S_Key = false
+
+  @Field(() => Boolean)
+  Up_Key = false
+
+  @Field(() => Boolean)
+  Down_Key = false
+}
+
+@ObjectType()
+export class PongPlayer {
+  constructor(nickname: string, id: string, presence: boolean) {
+    this.nickname = nickname
+    this.id = id
+    this.presence = presence
+  }
+
+  @Field(() => String)
+  id: string
+
+  @Field(() => String)
+  nickname: string
+
+  @Field(() => Int)
+  score = 0
+
+  @Field(() => Boolean)
+  presence: boolean
+
+  @Field(() => Controls)
+  controls = new Controls()
+}
+
```

Comment:
All @ObjectType, @Field... etc decorators are made for graphql, why is it usefull everywhere on your class ? (all of them)

Reply:
All the classe in game entity are returned to a user for the online game state, they all need to be specified as game type fo the graphql request, some @Field decorator will be deleted if they are not need in the request